### DISCUSSION TITLE
PostgreSQL type conversion

### DISCUSSION THREAD

**Discussion by crunchy234** [diesel-rs/diesel#3494] [Q&A] [answered]
I have a table that contains JSONB data and would like the query to return the JSON as Text in some cases.

The diesel definition of the table
```.rs
diesel::table! {
    user_data (id) {
        id -> Int4,
        data -> Jsonb,
    }
}
```

The SQL query I usually use is:
```.sql
SELECT id, data::text
FROM user_data;
```

I have tried looking in the `operators.rs` files to see if the `::` operator exists anywhere and can't seem to find anything.
Ideally, I'd like to write a query like the following.
```.rs
user_data::table.select((user_data::id, user_data::data.as_string())).load::<(i32, String)>(&mut db_connection)
```

Can you please explain how I could do something like this.

**Comment by weiznich** [comment, marked as answer]
Diesel does not provide a conversion operator out of the box. 
You can easily extend diesel to implement "custom" operators via the [`postfix_operator!`](https://docs.diesel.rs/2.0.x/diesel/macro.postfix_operator.html) macro. This allows you to define such custom extension methods:

```rust
diesel::postfix_operator!(ToText, "::text", diesel::sql_types::Text);

// use it directly via
        .select(ToText::new(users::id))

// or define a new extension trait/function 
```

**Reply by crunchy234** [reply]
Thanks!
